spec.html

```diff
@@ -7921,7 +7920,7 @@ <h1>ArraySetLength ( _A_, _Desc_ )</h1>
 
     <emu-clause id="sec-string-exotic-objects">
       <h1>String Exotic Objects</h1>
-      <p>A <em>String object</em> is an exotic object that encapsulates a String value and exposes virtual integer-indexed data properties corresponding to the individual code unit elements of the String value. String exotic objects always have a data property named `"length"` whose value is the number of code unit elements in the encapsulated String value. Both the code unit data properties and the `"length"` property are non-writable and non-configurable.</p>
```

Comment:
hm, i kind of feel like when it's a name, it should have quotes, and when it's referring to the property, it shouldn't?

Reply:
So "property named `"length"`" versus "the `length` property", you mean?

Any consistent approach is fine by me, though I would counterargue that in `let foo = 3;` we have an "identifier named `foo`" and no strings to be found. :smile:

Reply:
Because in the property case it’s actually a string :-)

Reply:
Right, I just mean that the verb 'named' shouldn't make a difference (_assuming I understood your concern correctly_)—I figure the two consistent approaches would be:
1. `"length"` in the algorithms and `length` in the exposition
2. `"length"` across the board

(...noting that the same ought to apply to `"prototype"`, `"constructor"`, etc.)

Reply:
Any of these consistent approaches is an improvement; let's see what the other editors think.

Reply:
I have a preference for quotes; right now we use `@@` to prefix well known symbols and the rest of properties are String values including integer indices; however, I can see how this could be confusing if you perform some lookups in algorithm by the value of a variable with name `length` if we did not distinguish `_length_` from `` `length` ``.  Since this is about consistency and not about distinguishing the values I'd lean towards @ljharb here and have it use quotes. It better represents what is going on. My main concern is actually about future key types being added and wanting to be sure we are clear about what the key types are when we are performing algorithm steps.

Reply:
That works—"quotes everywhere" is the most trivial to maintain anyway. :smile:

Reply:
...on the other hand, we're pretty close to (1) already, while (2) would be a more substantial shift—there are ~70 instances of `length`, ~70 of `prototype`, etc., not to mention that this also applies to methods like `toString`. Would we really want to change our policy on all of these?

Reply:
So I went through every instance of "length" in the spec and there are four situations:
1. length - an English word
2. _length_ - a formal parameter
3. `length` - a property, either referenced in prose (including assert text) or dot-accessed
4. `"length"` - a string literal passed as an argument (even in text like "a [[Get]] of `"length"`")

There were three outstanding cases outside of section 9, so I've added them to this PR.

If this is acceptable / helpful, I'd be happy to do similar consistency checks for other properties / methods as a follow-up...but if I'm just being a nuisance, please let me know. 😅